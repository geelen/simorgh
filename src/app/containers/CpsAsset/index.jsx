import React from 'react';
import compose from 'ramda/src/compose';
import CpsAssetMain from '../CpsAssetMain';

import withContexts from '../PageHandlers/withContexts';
import withPageWrapper from '../PageHandlers/withPageWrapper';
import withError from '../PageHandlers/withError';
import withLoading from '../PageHandlers/withLoading';
import withData from '../PageHandlers/withData';

const CpsAssetContainer = props => {
  return <CpsAssetMain {...props} />;
};

const EnhancedCpsAssetContainer = compose(
  withContexts,
  withPageWrapper,
  withLoading,
  withError,
  withData,
)(CpsAssetContainer);

export default EnhancedCpsAssetContainer;
