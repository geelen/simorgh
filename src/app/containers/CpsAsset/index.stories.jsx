import React from 'react';
import { storiesOf } from '@storybook/react';
import { inputProvider } from '@bbc/psammead-storybook-helpers';
import { withKnobs } from '@storybook/addon-knobs';

import CpsAsset from '.';
import indonesia from '../../../../data/indonesia/bbc_indonesian_radio/liveradio.json';
import korean from '../../../../data/korean/bbc_korean_radio/liveradio.json';
import tigrinya from '../../../../data/tigrinya/bbc_tigrinya_radio/liveradio.json';
import afaanoromoo from '../../../../data/afaanoromoo/bbc_afaanoromoo_radio/liveradio.json';
import amharic from '../../../../data/amharic/bbc_amharic_radio/liveradio.json';

const liveRadioFixtures = {
  indonesia,
  korean,
  tigrinya,
  afaanoromoo,
  amharic,
};

const matchFixtures = service => ({
  params: {
    mediaId: 'liveradio',
    serviceId: {
      indonesia: 'bbc_indonesian_radio',
      korean: 'bbc_korean_radio',
      tigrinya: 'bbc_tigrinya_radio',
      afaanoromoo: 'bbc_afaanoromoo_radio',
      amharic: 'bbc_amharic_radio',
    }[service],
  },
});

storiesOf('Pages|Cps Asset', module)
  .addDecorator(withKnobs)
  .add(
    'default',
    inputProvider({
      // eslint-disable-next-line react/prop-types
      componentFunction: ({ service }) => (
        <CpsAsset
          match={matchFixtures(service)}
          data={{
            pageData: liveRadioFixtures[service],
            status: 200,
          }}
          service={service}
          isAmp={false}
          loading={false}
          error=""
          pageType="media"
        />
      ),
      service: Object.keys(liveRadioFixtures),
      options: { defaultService: 'indonesia' },
    }),
  );
