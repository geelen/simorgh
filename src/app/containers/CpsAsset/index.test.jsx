import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import CpsAsset from '.';
import amharicPageData from '../../../../data/amharic/bbc_amharic_radio/liveradio';

const liveRadioScaffoldProps = {
  isAmp: false,
  pageType: 'media',
  service: 'amharic',
  pathname: '/pathname',
  match: {
    params: {
      serviceId: 'bbc_amharic_radio',
      mediaId: 'liveradio',
      pageData: amharicPageData,
    },
  },
  data: { status: 200 },
};

jest.mock('../PageHandlers/withPageWrapper', () => Component => {
  const PageWrapperContainer = props => (
    <div id="PageWrapperContainer">
      <Component {...props} />
    </div>
  );

  return PageWrapperContainer;
});

jest.mock('../PageHandlers/withLoading', () => Component => {
  const LoadingContainer = props => (
    <div id="LoadingContainer">
      <Component {...props} />
    </div>
  );

  return LoadingContainer;
});

jest.mock('../PageHandlers/withError', () => Component => {
  const ErrorContainer = props => (
    <div id="ErrorContainer">
      <Component {...props} />
    </div>
  );

  return ErrorContainer;
});

jest.mock('../PageHandlers/withData', () => Component => {
  const DataContainer = props => (
    <div id="DataContainer">
      <Component {...props} />
    </div>
  );

  return DataContainer;
});

jest.mock('../CpsAssetMain', () => {
  const CpsAssetMain = () => <div>CpsAssetMain</div>;

  return CpsAssetMain;
});

describe('Cps Asset', () => {
  describe('snapshots', () => {
    shouldMatchSnapshot(
      'should match scaffold snapshot',
      <CpsAsset {...liveRadioScaffoldProps} />,
    );
  });
});
