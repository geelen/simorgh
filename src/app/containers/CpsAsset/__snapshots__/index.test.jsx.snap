// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media Page snapshots should match scaffold snapshot 1`] = `
<div
  id="PageWrapperContainer"
>
  <div
    id="LoadingContainer"
  >
    <div
      id="ErrorContainer"
    >
      <div
        id="DataContainer"
      >
        <div>
          MediaPageMain
        </div>
      </div>
    </div>
  </div>
</div>
`;
